general/config.hpp

```diff
@@ -12,76 +12,89 @@
 // *****************************************************************************
 class config
 {
 private:
-   bool cuda = false;
    bool pa = false;
+   bool cuda = false;
+   bool occa = false;
    bool sync = false;
    bool nvvp = false;
    int dev;
    int gpu_count;
    CUdevice cuDevice;
    CUcontext cuContext;
    CUstream *cuStream;
+   OCCAdevice occaDevice;
+
 private:
+   // **************************************************************************
    config() {}
    config(config const&);
    void operator=(config const&);
+
 public:
-   static config& Get()
+   // **************************************************************************
+   static config& _Get()
    {
       static config singleton;
       return singleton;
    }
-   // **************************************************************************
-private:
-   void cuDeviceSetup(const int dev =0);
 
-public:
+private:
    // **************************************************************************
-   constexpr static inline bool nvcc()
-   {
-#ifdef __NVCC__
-      return true;
-#else
-      return false;
-#endif
-   }
+   void cudaDeviceSetup(const int =0);
+   void occaDeviceSetup();
+   void devSetup(const int =0);
 
+private:
    // **************************************************************************
-   inline bool Cuda() { return cuda; }
-   inline void Cuda(const bool mode) { cuda = mode; }
+   inline bool GetOcca() { return occa; }
+   inline void SetOcca(const bool mode) { occa = mode; }
+   inline OCCAdevice GetOccaDevice() { return occaDevice; }
 
    // **************************************************************************
-   inline bool PA() { return pa; }
-   inline void PA(const int mode) { pa = mode; }
+   inline bool GetCuda() { return cuda; }
+   inline void SetCuda(const bool mode) { cuda = mode;}
 
    // **************************************************************************
-   inline bool Sync(bool toggle=false) { return toggle?sync=!sync:sync; }
+   inline bool GetPA() { return pa; }
+   inline void SetPA(const int mode) { pa = mode; }
 
    // **************************************************************************
+   inline bool Sync(bool toggle=false) { return toggle?sync=!sync:sync; }
    inline bool Nvvp(bool toggle=false) { return toggle?nvvp=!nvvp:nvvp; }
 
    // **************************************************************************
-   void Setup() { cuDeviceSetup(); }
+   inline CUstream GetStream() { return *cuStream; }
 
+public:
    // **************************************************************************
-   inline CUstream Stream() { return *cuStream; }
+   // * Shortcuts
+   // **************************************************************************
+   static inline void Setup() { _Get().devSetup(); }
+   constexpr static inline bool Nvcc() { return cuNvcc(); }
+
+   static inline bool PA() { return _Get().GetPA(); }
+   static inline void PA(const bool b) { _Get().SetPA(b); }
```

Comment:
These methods aren't named like an action is being taken, maybe

```cpp
static inline void enablePA(const bool enabled = true) {
  _Get().SetPA(enabled);
}

static inline void enableCuda(const bool enabled = true) {
  _Get().SetCuda(enabled);
}

static inline void enableOcca(const bool enabled = true) {
  _Get().SetOcca(enabled);
}
```

Reply:
I kept the shortcuts whit the `Set` prefix for the action ones.